### PR TITLE
Port QSPI power mangement fixes from flash driver

### PR DESCRIPTION
**Pull request by krish2718**
* 95d867e8ed ("drivers/flash/nrf_qspi_nor: fix for missing device deactivation.")
  * da3ec7dbfa ("drivers: flash: Fix regression on nrf52840 anomaly 122 workaround")

Get lower power numbers when Wi-Fi is enabled (current consumption jumps from 2->7ma).

### CODE REVIEW

**Review by sachinthegreen — APPROVED** [review]

File: drivers/wifi/nrf700x/zephyr/src/qspi/src/qspi_if.c [hunk 1]
```diff
@@ -122,6 +115,10 @@ static inline int qspi_freq_to_sckfreq(int freq)
 #endif
 }
 
+static inline int qspi_dts_freq_to_sckfreq(void)
+{
+	return qspi_freq_to_sckfreq(INST_0_SCK_FREQUENCY);
```
Comment (leewkb4567) [inline drivers/wifi/nrf700x/zephyr/src/qspi/src/qspi_if.c:120]: The function qspi_freq_to_sckfreq() processes an input in MHz unit. Nevertheless, INST_0_SCK_FREQUENCY is specified in Hz. Therefore, a 24MHz in DTS will become 6MHz in this function...

File: drivers/wifi/nrf700x/zephyr/src/qspi/src/qspi_if.c [hunk 2]
```diff
@@ -113,6 +103,9 @@ struct qspi_nor_data {
 
 static inline int qspi_freq_to_sckfreq(int freq)
 {
+	if (freq > (NRF_QSPI_BASE_CLOCK_FREQ)) {
+		return NRF_QSPI_FREQ_DIV1;
```
Comment (leewkb4567) [inline drivers/wifi/nrf700x/zephyr/src/qspi/src/qspi_if.c:107]: This function input is given in MHz unit. Yet the macro NRF_QSPI_BASE_CLOCK_FREQ is specified in Hz.

**Review by leewkb4567 — CHANGES_REQUESTED** [review]

**Review by sr1dh48r — APPROVED** [review]

**Comment by krish2718** [comment]
I have tested this using PPK connected to 5340 and seeing about 2mA current consumption and also seeing that nrf7002 is able to wakeup the host from suspend and then nrf5340 is able to process and send ICMP replies.

File: drivers/wifi/nrf700x/zephyr/src/qspi/src/qspi_if.c [hunk 3]
```diff
@@ -111,17 +101,25 @@ struct qspi_nor_data {
...
 #else
 	/* QSPIM (2-32MHz): 32 MHz / (SCKFREQ + 1) */
-	return (32 / freq) - 1;
+	return (32 / freq_m) - 1;
 #endif
 }
 
+static inline int qspi_dts_freq_to_sckfreq(void)
+{
+	/* DTS freqyency is in HZ */
```
Comment (leewkb4567) [inline drivers/wifi/nrf700x/zephyr/src/qspi/src/qspi_if.c:120]:
```suggestion
	/* DTS frequency is in HZ */
```

File: drivers/wifi/nrf700x/zephyr/src/qspi/src/qspi_if.c [hunk 4]
```diff
@@ -388,22 +384,18 @@ static void qspi_handler(nrfx_qspi_evt_t event, void *p_context)
 		_qspi_complete(dev_data);
 }
 
-#if NRF52_ERRATA_122_PRESENT
 static bool qspi_initialized;
 
-static int anomaly_122_init(const struct device *dev)
+int qspi_device_init(const struct device *dev)
```
Comment (leewkb4567) [inline drivers/wifi/nrf700x/zephyr/src/qspi/src/qspi_if.c:389]:
```suggestion
static int qspi_device_init(const struct device *dev)
```

File: drivers/wifi/nrf700x/zephyr/src/qspi/src/qspi_if.c [hunk 5]
```diff
@@ -421,13 +414,10 @@ static int anomaly_122_init(const struct device *dev)
 	return ret;
 }
 
-static void anomaly_122_uninit(const struct device *dev)
+void qspi_device_uninit(const struct device *dev)
```
Comment (leewkb4567) [inline drivers/wifi/nrf700x/zephyr/src/qspi/src/qspi_if.c:417]:
```suggestion
static void qspi_device_uninit(const struct device *dev)
```

**Comment by krish2718** [comment]
> Please address the 3 comments.
> 
> Unfortunately, committing in GitHub won't pass the CI checks. It takes modifying the last commit.

It does work, I commit in to github, then fetch + rebase, but as I am using two laptops, so, might have accidentally overwritten the changes.

**Review by leewkb4567 — APPROVED** [review]

**Comment by leewkb4567** [comment]
> It does work, I commit in to github, then fetch + rebase, but as I am using two laptops, so, might have accidentally overwritten the changes.

I saw a commit done in GitHub did not have a description for CI checks. Anyway, it is good now.

**Pull request merged**